modules/container/filter.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package container
+
+import "slices"
+
+// FilterMapUnique ranges over the slice and calls include() for each element.
+// If the second returned value is true, the first returned value will be included in the resulting
+// slice (after deduplication).
+func FilterMapUnique[T, V comparable](elements []T, include func(T) (V, bool)) []V {
```

Comment:
Maybe `SliceUnique` or `UniqueSlice` is a better name?

Reply:
Renamed in https://github.com/go-gitea/gitea/pull/30339/commits/9b54f8f37badd88b0d2b29aeef9fbb113ac5c1cf